apps/prairielearn/elements/pl-figure/pl-figure.mustache

```diff
@@ -1,8 +1,8 @@
 {{^inline}}
-<div class="container-fluid">
-    <img src="{{src}}" {{#width}}width="{{width}}"{{/width}} alt="{{alt}}" class="img-fluid mx-auto d-block" />
+<div class="container-fluid {{#class}}{{class}}{{/class}}" title="{{title}}">
```

Comment:
Can you talk about your use case for `title`? It's difficult to discover, especially on touch devices.

Reply:
Depending on the use case, a [figure with a caption](https://getbootstrap.com/docs/4.6/content/figures/) might be a better alternative.

Reply:
@jcrayb could you give this question a look? This will help us determine a path forward for this PR.

Reply:
I think `<figure>` and `<figcaption>` would be the canonical way to show image attribution. It feels like a misuse of the `title` attribute to try to put authorship information there.

Reply:
I agree that we shouldn't support the use of `title` on images for any reason:
1. [MDN says](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#the_title_attribute):
    > The title attribute should also not be used as supplemental captioning information to accompany an image's alt description. If an image needs a caption, use the `figure` and `figcaption` elements.
2. [The HTML spec says](https://html.spec.whatwg.org/multipage/dom.html#the-title-attribute):
    > Relying on the `title` attribute is currently discouraged as many user agents do not expose the attribute in an accessible manner as required by this specification (e.g., requiring a pointing device such as a mouse to cause a tooltip to appear, which excludes keyboard-only users and touch-only users, such as anyone with a modern phone or tablet).

I like the idea of supporting `figure` and `figcaption` in general (not necessarily for author attribution).

Another possibility for author attribution might be something like:
```
<details>
  <summary>Image credit</summary>
  Image by Nathan Walters, licensed under CC BY 4.0.
</details>
```
or even perhaps:
```
<abbr title="Image by Nathan Walters, licensed under CC BY 4.0">Image credit</abbr>
```

@nwalters512 do you have other ideas about a standards-compliant way to put an expandable/tooltip/etc next to the figure to show credit?

Reply:
`<details>`/`<summary>` could work, thought I wouldn't bake this into `<pl-figure>`. I'm curious to hear your concerns about using `<figcaption>` for author attribution. I looked at news sites as a place where I'd probably find images with attribution and the three sites I checked all use `<figure>` and `<figcaption>`.

NYT:

<img width="738" alt="Screenshot 2024-07-24 at 10 15 07" src="https://github.com/user-attachments/assets/427227a0-c24e-4ce5-86c0-8a6ab830ed3a">


Washington Post:

<img width="653" alt="Screenshot 2024-07-24 at 10 14 39" src="https://github.com/user-attachments/assets/daf97df5-f760-432c-9706-972fa1253315">

The Verge (they actually wrap the citation in `<cite>`):

<img width="721" alt="Screenshot 2024-07-24 at 10 15 55" src="https://github.com/user-attachments/assets/de812d13-c9de-4d1f-b054-7449e191a00e">

If we add support for a `caption` attribute, we could easily recommend what the Washington Post does, just put author information in a parenthetical. If we wanted to provide different styles for the author information, I guess we could add support for a `citation` attribute or something? It's not clear to me that that's worth it.

Reply:
I don't see the need for an explicit `citation` attribute. I agree that supporting `figcaption` would be nice, both for this application and for other uses.

I also very much agree that we wouldn't bake-in support for `details`/`abbr`/etc for citations -- these are just ideas that could be used for anyone writing a question who wanted to add some text below a `pl-figure` (separate to the `pl-figure` element).